docs/Troubleshooting.md

```diff
@@ -138,3 +138,99 @@ and is a JSON object generated by the Quilt build system.
+        All As HAR**.
+        - Apple Safari: Click the **Export** button
+    1. Save the HAR-formatted file to your localhost.
+        ![Save browser Network error logs as HAR content](imgs/troubleshooting-logs-browser.png)
+1. CloudWatch logs for the Quilt cluster Elastic Container Service:
+    1. Install the [AWS CLI](https://aws.amazon.com/cli/) tool.
+    1. Export the AWS profile for your Quilt stack:
+        ```
+        export AWS_PROFILE=stage
+        ```
+    1. Execute the `cloudformation` command to retrieve your Quilt stack log group name:
+        ```
+        aws cloudformation describe-stack-resource --stack-name "YOUR_QUILT_STACK" --logical-resource-id "LogGroup"
+        ```
+        ```
+        {
+            "StackResourceDetail": {
+                "StackName": "YOUR_QUILT_STACK",
+                "StackId": "arn:aws:cloudformation:REGION:XXXXXXXXXXXX:stack/YOUR_QUILT_STACK/XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
+                "LogicalResourceId": "LogGroup",
+                "PhysicalResourceId": "YOUR_QUILT_STACK_LOG_GROUP_NAME",
+                "ResourceType": "AWS::Logs::LogGroup",
+                "LastUpdatedTimestamp": "2022-01-11T04:44:26.778Z",
+                "ResourceStatus": "UPDATE_COMPLETE",
+                "Metadata": "{}",
+                "DriftInformation": {
+                    "StackResourceDriftStatus": "IN_SYNC"
+                }
+            }
+        }
+        ```
+    1. Execute the following Bash script to save all Quilt ECS containers log entries for
+    the last 30 minutes to a tab-delimited text file `query_results.tsv`:
+        ```bash
+        #!/bin/bash
+
+        # Substitute the log group name from the aws cloudformation
+        # describe-stack-resource command above
+        log_group_name="YOUR_QUILT_STACK_LOG_GROUP_NAME"
+        file_out="query_results.tsv"
+
+        # Time now (in epoch seconds)
+        end_time=$(date +%s)
+        # Time 30 minutes ago (in epoch seconds)
+        start_time=$(date -v-30M +%s)
+
+        echo "end_time: $end_time"
+        echo "start_time: $start_time"
+        echo "log_group_name: $log_group_name"
+
+        # Define the query
+        query_string="fields @timestamp, @message | sort @timestamp desc"
+        query_id=$(aws logs start-query \
+            --log-group-name "$log_group_name" \
+            --start-time "$start_time" \
+            --end-time "$end_time" \
+            --query-string "$query_string" \
+            --output text \
+            --query 'queryId')
+ 
+        # Output the query id
+        echo "query_id: $query_id"
+ 
+        # Give the query time to run
+        echo "sleep for 10 seconds (let the query run)..."
+        sleep 10
+
+        echo "write results to $file_out"
+        query_results=$(aws logs get-query-results --query-id $query_id)
+        echo "$query_results" | jq -r '.results[] | [.[] | .value] | @tsv' | column -t > $file_out
+        ```
```

Comment:
OK for now, until we have a Python script (which will checked into source control, where we can use Python to get around OS specifics, etc.: Let's do something really simple like this instead, noting the date function may vary by OS (this is for Mac BSD), and the user may need to adjust the time range to capture their event.

```
aws logs filter-log-events \
  --log-group-name $YOUR_LOG_GROUP \
  --start-time "$(( $(date -u -v-30M +%s) * 1000 ))" \
  --end-time "$(( $(date -u +%s) * 1000 ))" > log-events.json
```

Reply:
Note the redirect at the end. All commands need to output into a file that's appropriately named.

Reply:
Oh, lastly, so that we can easily ask for and link to specific logs (not all customers need to send all logs), let's have a clean subsection for Browser Logs, CloudFormation Logs, ECS Logs, etc.